### PR TITLE
Internal OCS Controller cleanup

### PR DESCRIPTION
**Pull request by rullzer**
Just to see how CI handles it
- Allow to passon headers to an OCS response
- More logical split of V1/V2 response
- Fix tests

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -72,3 +72,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
```
Comment (nickvergessen) [inline .htaccess]: nop

File: lib/private/AppFramework/OCS/V1Response.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+		return $status;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function render() {
+		$meta = [
+			'status' => $this->getStatus() === Http::STATUS_OK ? 'ok' : 'failure',
```
Comment (nickvergessen) [inline lib/private/AppFramework/OCS/V1Response.php]: No, status is 'ok' for OCS code 100 ?

File: lib/public/AppFramework/OCSController.php [hunk 3]
```diff
@@ -72,6 +75,14 @@ public function __construct($appName,
 	}
 
 	/**
+	 * @param int $version
+	 * @since 9.2.0
+	 */
+	public function setOCSVersion($version) {
```
Comment (rullzer) [inline lib/public/AppFramework/OCSController.php]: I'm actually not to happy with this. But else we'll need yet another response type to convert to... I need to think about that.

Comment (rullzer) [inline lib/public/AppFramework/OCSController.php]: Reason is we can't just throw back a DataResponse as we need that to catch responses from other middleware.

Comment (nickvergessen) [inline lib/public/AppFramework/OCSController.php]: Mark as `@internal` so no one tries to use it.

**Comment by nickvergessen** [comment]
Works now with https://github.com/nextcloud/notifications/pull/21

:+1:

File: lib/public/AppFramework/OCSController.php (hunk 3 above)
Comment (rullzer) [inline lib/public/AppFramework/OCSController.php]: fair enough lets do that and then I will think about it some more

**Comment by rullzer** [comment]
Ok @nickvergessen @MorrisJobke @LukasReschke please have a look

**Comment by MorrisJobke** [comment]
Tested and OCS API still works 👍

**Pull request merged**